main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapActivity.java

```diff
@@ -850,6 +850,9 @@ public void onPause() {
 
     @Override
     protected void onResume() {
+        if (Settings.removeFromRouteOnLog()) {
+            viewModel.reloadIndividualRoute();
```

Comment:
how would this get triggered? (Where does the actual removal of RouteItem happen?)

Due to using the viewModel architecture you don't need an actual reload from database in `onResume`, you can simply remove the selected item from the `IndividualRoute` in memory, that should trigger the update on the map automatically. (But you still need to persist the removal, so calling `removeFirstMatchingIdFromIndividualRoute()` is still required.)